File: python/2.7/strategy/FlyBehavior.py
#import ABCMeta to use Abstract Class
from abc import ABCMeta, abstractmethod

class FlyBehavior:
	__metaclass__ = ABCMeta

	@abstractmethod
	def fly(self):
		pass

class FlyWithWings(FlyBehavior):
    def fly(self):
        print "I believe I can fly"

class FlyNoWay(FlyBehavior):
    def fly(self):
        print "I can't fly"

class FlyRocketPowered(FlyBehavior):
    def fly(self):
        print "I'm flying with a rocket"

File: php/5.3/strategy/com/duck/FlyBehavior.php
<?
namespace strategy;

interface FlyBehavior {
    public function fly();
}

class FlyWithWings implements FlyBehavior {
    public function fly() {
        echo "I believe I can fly\n";
    }
}

class FlyNoWay implements FlyBehavior {
    public function fly() {
        echo "I can't fly\n";
    }
}

class FlyRocketPowered implements FlyBehavior {
    public function fly() {
        echo "I'm flying with a rocket\n";
    }
}
?>
